### PR TITLE
ZFIN-7945 and ZFIN-7944

### PR DESCRIPTION
**Pull request by rtaylorzfin**
For NTR pages, this allows adding citations to the "chromosomal location" information.  It also displays the same information to the end users on NTR view pages.

### CODE REVIEW

**Comment by rtaylorzfin** [comment]
@cmpich , there are a couple places I'd appreciate your feedback:

1. I am introducing the isMarkerNtr variable to pass to the jsp for conditional rendering. I'm not sure if this is the most elegant way to do this.
2. I am storing citations for these "chromosomal locations" as RecordAttribution objects. I think this is the right way to go, but just wanted to check with you.  The entries would have a data_zdb_id that corresponds to the sequence_feature_chromosome_location table and a source_zdb_id to the publication.

File: source/org/zfin/marker/repository/HibernateMarkerRepository.java [hunk 1]
```diff
@@ -793,6 +794,57 @@ public void addDataAliasAttribution(DataAlias alias, Publication attribution, Ma
...
+
+            infrastructureRepository.insertUpdatesTable(genomeLocation.getZdbID(), "", "new attribution publication to genome location " + attributionZdbID + " to " + relZdbID);
+        }
+
+    }
+
+    public void synchronizeGenomeLocationAttributions(MarkerLocation genomeLocation, Set<String> publicationIDsToSync) {
+        //ADD
+        for (String publicationID : publicationIDsToSync) {
+            this.addGenomeLocationAttribution(genomeLocation, publicationID);
```
Comment (cmpich) [inline source/org/zfin/marker/repository/HibernateMarkerRepository.java:831]: do you need the 'this.' prefix to call the method in this class?

Comment (rtaylorzfin) [reply source/org/zfin/marker/repository/HibernateMarkerRepository.java:831]: Good point

**Review by cmpich — APPROVED** [review]
Is there no general controller method from the other UI components that creates an attribution for an entity, in your case a Location?

**Comment by rtaylorzfin** [comment]
> 

That's a good suggestion.  I looked through the code and didn't see anything that seemed like the right fit.

**Pull request merged**